_posts/2018-02-20--inception-pipelines-pt1.markdown

```diff
@@ -0,0 +1,99 @@
+
+The secret-sauce to the Inception Pipeline is the using the same CloudFormation stackname in the CLI/console invocation and the CodePipeline Action (CloudFormation parameter ```StageAdministerPipelineStackName```). Once the initial CloudFormation stack has been created, the CodePipeline action below maintains the pipeline.
+
+```yaml
+- Name: 'AdministerPipeline'
+    Actions:
+    - Name: 'AdministerPipeline'
+        ActionTypeId:
+            Category: 'Deploy'
+            Owner: 'AWS'
+            Provider: 'CloudFormation'
+            Version: '1'
+        Configuration:
+            ActionMode: 'REPLACE_ON_FAILURE' 
+            Capabilities: 'CAPABILITY_NAMED_IAM'
+            RoleArn: !GetAtt [CloudFormationDeployActionRole, Arn]
+            StackName: !Ref StageAdministerPipelineStackName
+            TemplateConfiguration: !Join ['', [!Ref RepositoryName, 'Source', '::aws_seed.json']]
+            TemplatePath: !Join ['', [!Ref RepositoryName, 'Source', '::aws_seed.yml']]
+        InputArtifacts:
+            - Name: !Join ['', [!Ref RepositoryName, 'Source']]
+        RunOrder: '1'
+```
+
+The CodePipeline is also configured to restart execution if it updates itself (```RestartExecutionOnUpdate: true```). This means when you add new stages, actions, etc, the pipeline will run the above action, realise its changed and then restart itself.
+
+While not strictly necessary, I would recommend that:
+
+1. The ```AdministerPipeline``` stage is only used for the ```AdministerPipeline``` action, i.e. don't add any other actions in.
+2. The ```AdministerPipeline``` stage is the first stage after the ```Source``` stage. This allows the pipeline to update itself before anything else runs.
+
+## Where do I get the seed files
+
+Go grab a copy from the [GitHub repository](https://github.com/MechanicalRock/InceptionPipeline/part1). I'll wait while you do.
+
+## What are the files
+
+|File|Description|
+|----|-----------|
+|init.sh|Executing this script (assuming the prerequisites are met) will create a seed pipeline |
+|aws_seed-cli-parameters.json|These are the paramerters to pass to the initial CloudFormation execution. These **MUST** match the values in ```aws_seed.json```|
+|aws_seed.json|These are the parameters used by the CloudFormation template when executed as a deployment action in CodePipeline|
+|aws_seed.yml|The pièce de résistance, the CloudFormation template that makes it all work|
+
+## Taking it for a spin
+
+1. Download the seed files from the [GitHub repository](https://github.com/MechanicalRock/InceptionPipeline/part1). Do not clone the repo, as the shell script will perform a ```git init```. Copy the downloaded files into a folder which will become your project folder.
```

Comment:
Broken link (as above)

Plus - point to the download zip directly to re-inforce the point.

It would be good to support git clone and remove the `.git` folder if necessary.

Reply:
Fixed broken link by pointing to the ZIP url.